From 1f99a88a30586831e212765ec73b1776270e2b4c Mon Sep 17 00:00:00 2001
From: Nicholas Devenish <ndevenish@gmail.com>
Date: Sat, 15 May 2021 12:13:11 +0100
Subject: [PATCH] Expose SWIG_PYTHON_STRICT_BYTE_CHAR as a variable

---
 pycbf/pycbf_i.w | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/pycbf/pycbf_i.w b/pycbf/pycbf_i.w
index fd1323f..849f101 100644
--- a/pycbf/pycbf_i.w
+++ b/pycbf/pycbf_i.w
@@ -394,6 +394,7 @@ static int convert_darray(PyObject *input, double *ptr, int size) {
 
 %{  // Here is the c code needed to compile the wrappers, but not 
     // to be wrapped 
+#include <stdbool.h>
 
 #include "../include/cbf.h"          
 #include "../include/cbf_simple.h"
@@ -478,4 +479,11 @@ void get_error_message(){
 // cbfhandle object
 %include "cbfhandlewrappers.i"
 
+// Expose the SWIG_PYTHON_STRICT_BYTE_CHAR build variable as a global
+%#ifdef SWIG_PYTHON_STRICT_BYTE_CHAR
+const bool SWIG_PYTHON_STRICT_BYTE_CHAR = 1;
+%#else
+const bool SWIG_PYTHON_STRICT_BYTE_CHAR = 0;
+%#endif
+
 @}

