diff --git a/pycbf/make_pycbf.w b/pycbf/make_pycbf.w
index 88b935e..0063e68 100644
--- a/pycbf/make_pycbf.w
+++ b/pycbf/make_pycbf.w
@@ -3448,8 +3448,8 @@ generic_wrapper = genericwrapper()
 
 
 def generate_wrappers(name_dict):
-   names = name_dict.keys()
-   for cname in names:
+   names = list(name_dict.keys())
+   for cname in sorted(names):
       prototype = name_dict[cname][0]
       docstring = name_dict[cname][1]
       # print "Generate wrappers: ", "::",cname,"::", prototype,"::", docstring
